docs/hosting/composer.md

```diff
@@ -0,0 +1,238 @@
+First, it's important to understand the concept of a "Composer project".
+
+If you are building a new PHP application from scratch with Composer, the
+first step is to create a `composer.json` file. As described in the Composer
+docs:
+
+> To start using Composer in your project, all you need is a `composer.json`
+> file. This file describes the dependencies of your project and may contain
+> other metadata as well.
+
+https://getcomposer.org/doc/01-basic-usage.md#composer-json-project-setup
+
+### Project templates
+
+Composer also provides a `create-project` command for starting a new project
+from an existing one. This allows for "project templates" to be provided for
+developers to build on top of.
+
+> You can use Composer to create new projects from an existing package. This
+> is the equivalent of doing a git clone/svn checkout followed by a
+> `composer install` of the vendors.
+
+https://getcomposer.org/doc/03-cli.md#create-project
+
+A good example of this is Drupal's `recommended-project` template, which is
+described as follows:
+
+> These project templates serve as a starting point for creating a
+> Composer-managed Drupal site. Once you have selected a template project and
+> created your own site, you will take ownership of your new project files.
+> Thereafter, future updates will be done with Composer.
+
+https://www.drupal.org/docs/develop/using-composer/starting-a-site-using-drupal-composer-project-templates
+
+For example, to create a new project based on Drupal's `recommended-project`
+template:
+
+    composer create-project drupal/recommended-project
+
+This will initialize a `composer.json` and `composer.lock` file that serves as
+a template for a Drupal website.
+
+### farmOS project template
+
+farmOS provides its own project template:
+https://packagist.org/packages/farmOS/project
+
+The official farmOS releases are built from this template. It can also be used
+to start your own project based on farmOS. This process is described in the
+next section.
+
+## Starting your farmOS project
+
+To start your own project based on the farmOS project template, open a new
+directory and run the following command:
+
+    composer create-project farmos/project
+
+This will create a `composer.json` file in your directory, copied directly from
+the [farmOS project template](https://packagist.org/packages/farmos/project).
+
+You can use this `composer.json` as the starting point for your own project's
+code repository with [Git](https://git-scm.com/):
+
+    git init
+    git add composer.json
+    git commit -m 'Intitial commit of my farmOS project.'
+
+Open `composer.json` in a text editor and change the metadata (eg: `name`,
+`description`, `homepage`, etc) to customize it for your project.
+
+It is important to understand that your project is *not* farmOS, but rather
+farmOS is a *dependency* of your project. You can see this clearly by observing
+that `farmos/farmos` is included in your `composer.json` file's `require`
+section.
```

Comment:
As a reader, it's not very clear **why** it is important to understand this... I'm guessing we just want them to name it something other than "farmOS" to avoid conflicts.

Reply:
Yea I see what you mean. I rearranged the text to try to make that more clear:

https://github.com/farmOS/farmOS/pull/648/commits/8d0149d561169af26ad7bb3e3f613697e17eea42